app/services/placements/organisation_finder.rb

```diff
@@ -0,0 +1,24 @@
+class Placements::OrganisationFinder
+  include ServicePattern
+
+  def initialize(filters: [], search_term: "", _page: 1)
+    @filters = filters
+    @search_term = search_term
+  end
+
+  def call
+    if filters.blank?
+      organisations
+    else
+      organisations.where(organisation_type: filters)
+    end
+  end
+
+  private
+
+  attr_reader :filters, :search_term
+
+  def organisations
+    @organisations ||= search_term.blank? ? PgSearch::Document.all.order(:content) : PgSearch.multisearch(search_term)
```

Comment:
I'm not very familiar with PG documents and this works for now but will this return documents from another model if we decide to use it for users for example? What I'm trying to ask here is should we get all documents by `organisation_type` for example? Or is the pg document table only for searching organisations and if we want to implement this somewhere else we create a different table?

Reply:
Good point